accounting/accounting.go

```diff
@@ -36,52 +34,18 @@ type State struct {
 }
 
 // New creates a new State object.
-func New(db idb.IndexerDb) *State {
-	return &State{db: db, defaultFrozen: make(map[uint64]bool)}
-}
-
-func (accounting *State) getTxnCounter(round uint64) (txnCounter uint64, err error) {
-	if round != accounting.txnCounterRound {
-		block, err := accounting.db.GetBlock(round)
-		if err != nil {
-			return 0, err
-		}
-		accounting.txnCounter = block.TxnCounter
-		accounting.txnCounterRound = round
-	}
-	return accounting.txnCounter, nil
+func New() *State {
+	return &State{defaultFrozen: make(map[uint64]bool)}
 }
 
-func (accounting *State) initRound(round uint64) error {
-	block, err := accounting.db.GetBlock(round)
-	if err != nil {
-		return err
-	}
+func (accounting *State) InitRound(block types.Block) error {
 	accounting.feeAddr = block.FeeSink
 	accounting.rewardAddr = block.RewardsPool
 	accounting.rewardsLevel = block.RewardsLevel
-	accounting.currentRound = round
-	return nil
-}
-
-func (accounting *State) commitRound() error {
-	if !accounting.dirty {
-		return nil
-	}
-	err := accounting.db.CommitRoundAccounting(accounting.RoundUpdates, accounting.currentRound, accounting.rewardsLevel)
-	if err != nil {
-		return err
-	}
-	accounting.RoundUpdates.Clear()
-	accounting.dirty = false
```

Comment:
looks like we lost these two lines in the shuffle

Reply:
Thanks, missing the round update clear is probably why the tests failed.
I think that `accounting.dirty` is no longer needed since I'm checking the payset length in importer now.